src/meta-srv/src/error.rs

```diff
@@ -540,6 +540,13 @@ pub enum Error {
     #[snafu(display("Expected to retry later, reason: {}", reason))]
     RetryLater { reason: String, location: Location },
 
+    #[snafu(display("Expected to retry later, reason: {}", reason))]
+    RetryLaterWithSource {
```

Comment:
Can we get rid of the old `RetryLater` and make `RetryLaterWithSource` the new `RetryLater` error? It seems all retry later errors should have sources.

Reply:
Not all retries have an error inside; some are just error messages. This is why I add a  indirection layer before.